libyul/backends/wasm/EVMToEWasmTranslator.cpp

```diff
@@ -342,22 +347,55 @@ function u256_to_i32(x1, x2, x3, x4) -> v {
 	v := x4
 }
 
+function u256_to_byte(x1, x2, x3, x4) -> v {
+	if i64.ne(0, i64.or(i64.or(x1, x2), x3)) { invalid() }
+	if i64.gt_u(x4, 255) { invalid() }
+	v := x4
+}
+
 function u256_to_i32ptr(x1, x2, x3, x4) -> v {
 	v := u256_to_i32(x1, x2, x3, x4)
 }
 
+function to_internal_i32ptr(x1, x2, x3, x4) -> r {
+	let p := u256_to_i32ptr(x1, x2, x3, x4)
+	r := i64.add(p, 64)
+	if i64.lt_u(r, p) { invalid() }
+}
+
+function u256_to_address(x1, x2, x3, x4) -> r1, r2, r3 {
+	if i64.ne(0, x1) { invalid() }
+	if i64.ne(0, i64.shr_u(x2, 32)) { invalid() }
+	r1 := x2
+	r2 := x3
+	r3 := x4
+}
+
 function keccak256(x1, x2, x3, x4, y1, y2, y3, y4) -> z1, z2, z3, z4 {
-	// TODO implement
-	unreachable()
+	mstore_address(0, 0, 0, 0, 0x09)
```

Comment:
I'd say to postpone this one in a different PR because it requires implementation in EVMHost too.